superset/models/core.py

```diff
@@ -87,6 +87,37 @@ def copy_dashboard(mapper, connection, target):
     session.commit()
 
 
+def query_obj_backfill(form_data, datasource_id=None, datasource_type=None):
+    # Backfill query_obj. This will be generated on the client in the future.
+    # Note we unpack the metrics on the viz here so we can create the metrics
+    # dictionary during viz initialization without leveraging form_data.
+    #
+    # TODO: Note structure of the backfilled query_obj is temporary. It will be
+    # restructured as we build out the query object on the client side.
+    # The fields included are necessary to load a table viz without form_data
+    query_obj = {
+        'viz_type': form_data.get('viz_type', 'table'),
```

Comment:
hmm, the downstream method, `get_viz` (which `get_payload` is called from) needs to know the `viz_type` in order to initialize, so we'll have to provide `viz_type` somehow. We could pass in `viz_type` as a parameter to `/api/v1/query`. Thoughts?

Reply:
I agree it should be `viz_type` because otherwise charts will be coupled with the backend, which is what we're trying to move away from for the `ChartPlugin`s.

Reply:
Eventually, in viz.py (if we still keep it), we will only have the BaseVis class. So here, actually we can first try the visualization type WorldCloud and BigNumber, since the only thing left for this two viz classes is query object logic.